compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -414,11 +416,8 @@ public BallerinaTypeDefinitionSymbol createTypeDefinition(BSymbol typeSymbol, St
         }
 
         if (typeSymbol.kind == SymbolKind.TYPE_DEF) {
-            for (BLangAnnotationAttachment annAttachment : ((BTypeDefinitionSymbol) typeSymbol).annAttachments) {
-                if (annAttachment.annotationSymbol == null) {
-                    continue;
-                }
-                symbolBuilder.withAnnotation(createAnnotationSymbol(annAttachment.annotationSymbol));
+            for (AnnotationAttachmentSymbol annAttachment : ((BTypeDefinitionSymbol) typeSymbol).getAnnotations()) {
+                symbolBuilder.withAnnotation(createAnnotationSymbol((BAnnotationAttachmentSymbol) annAttachment));
```

Comment:
Is having an annot attachment symbol always guaranteed? Even for incomplete/incorrect code? For example, what would be the behaviour if someone adds an annotation attachment using a non-existent annotation?

Reply:
Atm, the annotation attachment symbol is created only if there are no errors - https://github.com/ballerina-platform/ballerina-lang/pull/35189/files#diff-8aa3f7b695ef5b4ee6addb351eeac82a5ceda03b984a181d742b8f5727bf1f25R961. So for invalid cases, `BTypeDefinitionSymbol`  and other symbols wouldn't have an attachment symbol for that symbol.

Reply:
It'll have those of the correct attachments. Just that it won't have anything for wrong attachments.